kafka_consumer/datadog_checks/kafka_consumer/client/confluent_kafka_client.py

```diff
@@ -1,14 +1,76 @@
 # (C) Datadog, Inc. 2023-present
 # All rights reserved
 # Licensed under a 3-clause BSD style license (see LICENSE)
+from confluent_kafka.admin import AdminClient
+from confluent_kafka import Consumer, TopicPartition
 
+EXCLUDED_TOPICS = ['__consumer_offsets', '__transaction_state']
 
 class ConfluentKafkaClient:
-    def __init__(self) -> None:
-        pass
+    def __init__(self, config, tls_context, log) -> None:
+        self.config = config
+        self.log = log
+        self._kafka_client = None
+        self._highwater_offsets = {}
+        self._consumer_offsets = {}
+        self._tls_context = tls_context
+    
+    @property
+    def kafka_client(self):
+        if self._kafka_client is None:
+            # self.conf is just the config options from librdkafka
+            self._kafka_client = AdminClient(self.conf)
+
+        return self._kafka_client
+
+    def get_highwater_offsets(self):
+        # {(topic, partition): offset}
+        for consumer_group in self.config._consumer_groups:
+            config = {
+                "bootstrap.servers" : self.config._kafka_connect_str,
+                "group.id": consumer_group,
+            }
+            consumer = Consumer(config)
+
+            # TODO: AdminClient also has a list_topics(), see if Consumer.list_topics() is the same
```

Comment:
AdminClient's `list_topics()` does the same thing as Consumer's: https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#id12